api/spec/src/productcatalog/routes.tsp

```diff
@@ -0,0 +1,385 @@
+namespace OpenMeter.ProductCatalog;
+
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+@route("/api/v1/plans")
+@tag("Plan (Experimental)")
+interface PlansById {
+  /**
+   * List all plans.
+   */
+  @get
+  @operationId("listPlans")
+  @summary("List plans")
+  list(
+    /**
+     * Include deleted plans in response.
+     *
+     * Usage: `?includeDeleted=true`
+     */
+    @query
+    @example(true)
+    includeDeleted?: boolean = false,
+
+    /**
+     * Filter by plan.id attribute
+     */
+    @query(#{ explode: true })
+    id?: string[],
+
+    /**
+     * Filter by plan.key attribute
+     */
+    @query(#{ explode: true })
+    key?: string[],
+
+    /**
+     * Filter by plan.key and plan.version attributes
+     */
+    @query(#{ explode: true, style: "deepObject" })
+    keyVersion?: Record<integer>,
```

Comment:
Are these all independent filters? 
e.g. `key=key-1&keyVersion[key-2]=v3` returns nothing, or would they have precedence?

Reply:
So they are independent filters. Like `WHERE plan.id = 'x' OR plan.key = 'key-1' OR (plan.key='key-2' AND plan.version = 3)`